release_notes.py

```diff
@@ -9,301 +9,337 @@
-                    section_body += newStr
-            if len(self.modifiedStore) > 0:
-                modifiedStr = ""
-                modified_count = 0
-                for rawContent in self.modifiedStore:
-                    cnt = self.loadData(rawContent)
-                    ans = self.modifiedReleaseNotes(cnt)
-                    if ans is None:
-                        print_error(cnt["name"] + " is missing releaseNotes entry")
-                        missingReleaseNotes = True
-                    elif ans is not "":
-                        modifiedStr += ans
-                        modified_count += 1
-                if len(modifiedStr) > 0:
-                    if modified_count > 1:
-                        section_body += "\n#### " + str(modified_count) + " Improved " + self.getHeader() + "\n"
-                    else:
-                        section_body += "\n#### Improved " + self.getHeader() + "\n"
-                    section_body += modifiedStr
-            if len(self.deletedStore) > 0:
-                section_body += "\n#### Removed " + self.getHeader() + "\n"
-                for rawContent in self.deletedStore:
-                    section_body += "- " + rawContent + "\n"
-
-            if missingReleaseNotes:
+                        count_str = " " + str(new_count)
+
+                    res = "\n####%s%s %s\n" % (count_str, title_prefix, self.get_header())
+                res += new_str
+
+        if missing_rn:
+            return None
+
+        return res
+
+    def generate_release_notes(self):
+        res = ""
+
+        if len(self.modified_store) + len(self.deleted_store) + len(self.added_store) > 0:
+            print "starting %s RN" % (self.get_header(),)
+
+            # Added files
+            add_rn = self.release_notes_section(self.added_store, NEW_RN)
+
+            # Modified files
+            modified_rn = self.release_notes_section(self.modified_store, MODIFIED_RN)
+
+            if add_rn is None or modified_rn is None:
                 return None
+
+            section_body = add_rn + modified_rn
+
+            # Deleted files
+            if len(self.deleted_store) > 0:
+                section_body += "\n##### Removed " + self.get_header() + "\n"
+                for name in self.deleted_store:
+                    print " - adding release notes (Removed) for - [%s]" % (name,),
+                    section_body += "- __" + name + "__\n"
+                    print "Success"
+
             if len(section_body) > 0:
-                res = "### " + self.getHeader() + "\n"
+                res = "### " + self.get_header() + "\n"
                 res += section_body
 
         return res
 
 
 class ScriptContent(Content):
-
-    def loadData(self, data):
+    def load_data(self, data):
         return yaml.safe_load(data)
 
-    def getHeader(self):
+    def get_header(self):
         return "Scripts"
 
-    def addedReleaseNotes(self, cnt):
+    def added_release_notes(self, cnt):
```

Comment:
why is this different from other implementation of the function in other classes (PlaybookContent, ReportContent...)

Reply:
Do you mean script release notes?

taken from script comment (equivalent to description)